docs/git-101.md

```diff
@@ -0,0 +1,129 @@
+
+See those guidelines and best practices for details: [How to create a good commit message.](https://chris.beams.io/posts/git-commit/)
+
+After you are finished with creating a message, save and close your editor, and the commit will be created.
+
+You can create multiple commits on your branch when you feel it's necessary to divide your work into smaller chunks, or you just like working in small steps.
+
+Some useful commands:
+- `$ git add -u` - add all modified and tracked files
+- `$ git add .` - add all files in the current directory (also unversioned ones that are not included in the .gitignore file)
+- `$ git commit -v` - commit with the preview of made modifications. This preview will NOT be included in the commit message, it's just a way to help developer rememeber what he's actually changed
+- `$ git commit -m 'Change code so that it works better'` - create commit with commit message in one take
+
+## Publish your changes
+
+In order to publish your changes you need to push your branch to the remote repository:
+```
+$ git checkout your-feature-branch-123
+$ git push origin your-feature-branch-123
+```
+
+It is forbidden to publish your changes directly to a base branch, the only way is to publish your changes through a separate branch, and then have it merged after the whole review process.
+
+## Creating Pull request
+
+Pull requests are the important way of communication, so they should be used well.
+
+- Create pull request for every feature / bug fix you have implemented.
+- Create pull request based on your branch, which was compared to the base branch.
+- It is your responsibility to keep your pull request up-to-date, so when there are any new changes on the base branch before your pull request is closed, you should update your branch to include those changes
+- Pull request is an opportunity for the team to do a code review and discuss this solution and also for the CI to perform automated tests, which will verify the commits
+
+### How to create
+
+Just create it from the GitHub website, comparing your branch with `develop`. Or use some IDE plugin integrating well with GitHub, if you want to.
+
+### Updating code in your pull request
+
+If you need to make some changes (tests are failing, or there were changes requested by the reviewer, etc.) just put additional commits to your branch. They will be automatically added to this currently opened pull request.
+
+### Updating pull request to the newest changes on the base branch
+
+Every time your PR is behind current tip of the base branch, it's your responsibility to update it.
+
+**NEVER do it through GitHub website, as it creates merge commits and messes up the history!**
+
+To update pull request follow the procedure from the section above on getting the latest version of the code.
+After you are done rebasing you need to push your updated branch. But because the whole branch has changed, you need to rewrite it:
+```
+$ git push origin your-feature-branch-123 --force
+```
```

Comment:
bardzo pomógł mi [ten dokument](https://zwischenzugs.com/2018/03/14/five-key-git-concepts-explained-the-hard-way/) polecam